mongo/src/main/java/org/axonframework/mongo/eventsourcing/eventstore/AbstractMongoEventStorageStrategy.java

```diff
@@ -192,6 +192,15 @@ public void deleteSnapshots(MongoCollection<Document> snapshotCollection, String
         return stream(cursor.spliterator(), false).findFirst().map(this::extractSnapshot);
     }
 
+    @Override
+    public Stream<? extends DomainEventData<?>> findSnapshots(MongoCollection<Document> snapshotCollection,
+                                                              String aggregateIdentifier) {
+        FindIterable<Document> cursor =
+                snapshotCollection.find(eq(eventConfiguration.aggregateIdentifierProperty(), aggregateIdentifier))
+                                  .sort(new BasicDBObject(eventConfiguration.sequenceNumberProperty(), ORDER_ASC)).limit(1);
```

Comment:
I am by no means a Mongo expert, but wont the `limit(1)` call ensure you'll only get one snapshot?

Reply:
Nice catch, was copying `findLastSnapshot` method logic.